kite-data/kite-data-spark/src/main/scala/com/databricks/spark/avro/SchemaSupport.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2014 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.spark.avro
+
+import org.apache.avro.{Schema, SchemaBuilder}
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.types.{DataType, StructType}
+
+trait SchemaSupport {
```

Comment:
Unfortunately not, the databricks spark-avro project provides mechanisms for reading/writing data frames form/to avro files. It internally uses classes that convert data frame schemas to AVRO schemas and vice versa. Unfortunately either they are private or they are private at the package level. I wanted to reuse as much I could what already done in that project this is why I had to introduce that hacks and to put something under this package. It's ugly I understand but I couldn't do otherwise.